### ISSUE TITLE
custom training on spaCy’s PoS generates _SP for all tokens

### ISSUE THREAD

**Issue by HsengivS** [explosion/spaCy#6241]
Hello reader, 

I am trying to build a custom pos tagging using Spacy, with the provided sample code example from spacy's documentation. I am getting the same _SP tag for all tokens as shown below, Kindly assist me with this issue.

##  Environment

* Operating System: Windows 10
* Python Version Used: Python 3.7.6
* spaCy Version Used: 2.3.0

My code:
```
from __future__ import unicode_literals, print_function
import plac
import random
from pathlib import Path
import spacy
from spacy.util import minibatch, compounding

TAG_MAP = {"N": {"pos": "NOUN"}, "V": {"pos": "VERB"}, "J": {"pos": "ADJ"}}

TRAIN_DATA = [
    ("I like green eggs", {"tags": ["N", "V", "J", "N"]}),
    ("Eat blue ham", {"tags": ["V", "J", "N"]}),
]

@plac.annotations(
    lang=("ISO Code of language to use", "option", "l", str),
    output_dir=("Optional output directory", "option", "o", Path),
    n_iter=("Number of training iterations", "option", "n", int),
)
def main(lang="en", output_dir=None, n_iter=25):
    nlp = spacy.blank(lang)
    tagger = nlp.create_pipe("tagger")
    for tag, values in TAG_MAP.items():
        tagger.add_label(tag, values)
    nlp.add_pipe(tagger)

    optimizer = nlp.begin_training()
    for i in range(n_iter):
        random.shuffle(TRAIN_DATA)
        losses = {}
        batches = minibatch(TRAIN_DATA, size=compounding(4.0, 32.0, 1.001))
        for batch in batches:
            texts, annotations = zip(*batch)
            nlp.update(texts, annotations, sgd=optimizer, losses=losses)
        print("Losses", losses)


    test_text = "I like blue eggs"
    doc = nlp(test_text)
    print("Tags", [(t.text, t.tag_, t.pos_) for t in doc])

    # save model to output directory
    if output_dir is not None:
        output_dir = Path(output_dir)
        if not output_dir.exists():
            output_dir.mkdir()
        nlp.to_disk(output_dir)
        print("Saved model to", output_dir)

        # test the save model
        print("Loading from", output_dir)
        nlp2 = spacy.load(output_dir)
        doc = nlp2(test_text)
        print("Tags", [(t.text, t.tag_, t.pos_) for t in doc])


if __name__ == "__main__":
    plac.call(main)
```

Output:

```
UserWarning: [W022] Training a new part-of-speech tagger using a model with no lemmatization rules or data. This means that the trained model may not be able to lemmatize correctly. If this is intentional or the language you're using doesn't have lemmatization data, please ignore this warning. If this is surprising, make sure you have the spacy-lookups-data package installed.
  **kwargs
C:\ProgramData\Anaconda3\envs\spacy_2_3_0\lib\site-packages\spacy\language.py:636: UserWarning: [W033] Training a new part-of-speech tagger using a model with no lexeme normalization table. This may degrade the performance of the model to some degree. If this is intentional or
the language you're using doesn't have a normalization table, please ignore this warning. If this is surprising, make sure you have the spacy-lookups-data package installed. The languages with lexeme normalization tables are currently: da, de, el, en, id, lb, pt, ru, sr, ta, th.
  **kwargs
Losses {'tagger': 0.0}
Losses {'tagger': 0.0}
Losses {'tagger': 0.0}
Losses {'tagger': 0.0}
Losses {'tagger': 0.0}
Losses {'tagger': 0.0}
Losses {'tagger': 0.0}
Losses {'tagger': 0.0}
Losses {'tagger': 0.0}
Losses {'tagger': 0.0}
Losses {'tagger': 0.0}
Losses {'tagger': 0.0}
Losses {'tagger': 0.0}
Losses {'tagger': 0.0}
Losses {'tagger': 0.0}
Losses {'tagger': 0.0}
Losses {'tagger': 0.0}
Losses {'tagger': 0.0}
Losses {'tagger': 0.0}
Losses {'tagger': 0.0}
Losses {'tagger': 0.0}
Losses {'tagger': 0.0}
Losses {'tagger': 0.0}
Losses {'tagger': 0.0}
Losses {'tagger': 0.0}
Tags [('I', '_SP', 'SPACE'), ('like', '_SP', 'SPACE'), ('blue', '_SP', 'SPACE'), ('eggs', '_SP', 'SPACE')]
```

**Comment by adrianeboyd** [comment]
I think this is a bug that's fixed in v2.3.1. Can you try v2.3.1 or v2.3.2 (the most recent release)?

**Comment by HsengivS** [comment]
Thank you for assisting.

The issue got solved by updating the spacy version
